src/Http/Http.Abstractions/src/HttpContext.cs

```diff
@@ -68,7 +68,7 @@ public abstract class HttpContext
         /// <summary>
         /// Gets or sets the object used to manage user session data for this request.
         /// </summary>
-        public abstract ISession? Session { get; set; }
+        public abstract ISession Session { get; set; }
```

Comment:
I thought the feature throws if it hasn't been set up.

Reply:
DefaultHttpContext throws if the feature is missing, but it doesn't ensure anything about the actual ISession value.

Reply:
I guess. But that's true of nearly all of the features, no? We don't verify that the feature has been configured weirdly. In theory, most features could be implemented to return null values, but it would be incredibly cumbersome to use these APIs if they were all marked as nullable. In the ordinary case, it's either non-null or throws, no?

Reply:
I think we need to apply some pragmatism with nullable attributes.

Yes something could technically be null if an app was setup to do an unusual thing, but I think we need to focus on the 99% usage.

Reply:
The question is if these annotations are trying to communicate "SHOULD NOT" or "MUST NOT" be null (RFC terms). If we stick with marking the common nullable cases that should help people avoid common mistakes.

If we find later we're too loose or too strict with these annotations, how breaking is it to change them? Doesn't it introduce new warnings either way?